codecov_cli/helpers/ci_adapters/cloudbuild.py

```diff
@@ -0,0 +1,42 @@
+import os
+
+from codecov_cli.helpers.ci_adapters.base import CIAdapterBase
+
+
+class GoogleCloudBuildCIAdapter(CIAdapterBase):
+    # https://cloud.google.com/build/docs/configuring-builds/substitute-variable-values
+
+    def detect(self) -> bool:
```

Comment:
Add a comment about detection, since it will not work without setting this.
```suggestion
    def detect(self) -> bool:
        """By default Cloud Build exports no environment variables, so this requires `automapSubstiutions: true` to be set in cloudbuild.yaml"""
```